src/transformers/models/seggpt/image_processing_seggpt.py

```diff
@@ -0,0 +1,627 @@
+    get_channel_dimension_axis,
+    infer_channel_dimension_format,
+    is_scaled_image,
+    make_list_of_images,
+    to_numpy_array,
+    valid_images,
+)
+from ...utils import TensorType, is_torch_available, logging, requires_backends
+
+
+if is_torch_available():
+    import torch
+
+
+logger = logging.get_logger(__name__)
+
+
+# See https://arxiv.org/pdf/2212.02499.pdf  at 3.1 Redefining Output Spaces as "Images" - Semantic Segmentation from PAINTER paper
+# Taken from https://github.com/Abdullah-Meda/Painter/blob/main/Painter/data/coco_semseg/gen_color_coco_panoptic_segm.py#L31
+def build_palette(num_labels: int) -> List[Tuple[int, int]]:
+    base = int(num_labels ** (1 / 3)) + 1
+    margin = 256 // base
+
+    # we assume that class_idx 0 is the background which is mapped to black
+    color_list = [(0, 0, 0)]
+    for location in range(num_labels):
+        num_seq_r = location // base**2
+        num_seq_g = (location % base**2) // base
+        num_seq_b = location % base
+
+        R = 255 - num_seq_r * margin
+        G = 255 - num_seq_g * margin
+        B = 255 - num_seq_b * margin
+
+        color_list.append((R, G, B))
+
+    return color_list
+
+
+def get_num_channels(image: np.ndarray, input_data_format: ChannelDimension) -> int:
+    if image.ndim == 2:
+        return 1
+
+    channel_idx = get_channel_dimension_axis(image, input_data_format)
+    return image.shape[channel_idx]
+
+
+def mask_to_rgb(
+    mask: np.ndarray,
+    palette: Optional[List[Tuple[int, int]]] = None,
+    input_data_format: Optional[ChannelDimension] = None,
+    data_format: Optional[ChannelDimension] = None,
+) -> np.ndarray:
+    if input_data_format is None:
+        input_data_format = infer_channel_dimension_format(mask) if mask.ndim > 2 else ChannelDimension.LAST
+
+    num_channels = get_num_channels(mask, input_data_format)
+
+    if num_channels == 3:
+        return to_channel_dimension_format(mask, data_format, input_data_format) if data_format is not None else mask
+
+    mask = mask[0] if mask.ndim == 3 else mask
+
+    if palette is not None:
+        height, width = mask.shape
+
+        if input_data_format == ChannelDimension.LAST:
+            rgb_mask = np.zeros((height, width, 3), dtype=np.uint8)
+        else:
+            rgb_mask = np.zeros((3, height, width), dtype=np.uint8)
+
+        classes_in_mask = np.unique(mask)
+
+        for class_idx in classes_in_mask:
+            rgb_value = palette[class_idx]
+            class_mask = (mask == class_idx).astype(np.uint8)
+            class_mask = np.expand_dims(class_mask, axis=-1)
+            class_rgb_mask = class_mask * np.array(rgb_value)
+            rgb_mask += class_rgb_mask.astype(np.uint8)
+
+        rgb_mask = np.clip(rgb_mask, 0, 255).astype(np.uint8)
+    else:
+        if input_data_format == ChannelDimension.FIRST:
+            rgb_mask = np.repeat(mask[None, ...], 3, axis=0)
+        else:
+            rgb_mask = np.repeat(mask[..., None], 3, axis=-1)
```

Comment:
Same comment here. 

Instead - the condition should be on `data_format` which controls the output data format of the mask/image

```suggestion
        
        rgb_mask = np.repeat(mask[None, ...], 3, axis=0)
    
    if data_format is not None: 
        rgb_mask = to_channel_dimension_format(rgb_mask, data_format, ChannelDimension.FIRST)
```

Reply:
You're right, I was seeing ghosts 👻  (2) https://github.com/huggingface/transformers/pull/27735/commits/3373cbb63e5c3b364027b54b2bbcfbd4433f9d25